documentation/docs/decentralized_identifiers/resolve.mdx

```diff
@@ -7,18 +7,152 @@ keywords:
 - Resolve
 ---
 import CodeSnippet from '../../src/components/CodeSnippetComponent'
-import resolution from  '!!raw-loader!../../../examples/low-level-api/resolution.rs';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
 
-TODO: Explain how resolving works including arguments.
 
-## Example
+DID resolution is the process of fetching a [DID Document](https://www.w3.org/TR/did-core/#dfn-did-documents) corresponding to a given [DID](https://www.w3.org/TR/did-core/#dfn-decentralized-identifiers). 
+The [IOTA Identity Framework](https://github.com/iotaledger/identity.rs) supports resolving DID Documents that are stored on an IOTA Tangle (public or private). The main tool supplied 
+by the IOTA Identity Framework to handle DID Document resolution in a type safe manner is the `Resolver`. A DID Resolver as defined in the [W3C Decentralized Identifiers specification](https://www.w3.org/TR/did-core/#dfn-did-resolvers)
+enforces the signature of the resolution function in a manner that is more centered around Web/API resolution rather than a strongly typed framework. This is the reason why the `Resolver` provided by the IOTA Identity Framework deviates somewhat from 
+the W3C specification.
 
-A short example that shows how you can resolve a DID Document. This example returns the latest DID Document.
 
-<CodeSnippet 
-    nodeReplitLink="https://repl.it/@IOTAFoundation/resolution?lite=true"
-    rustContent={resolution}
-    nodeGithubLink = "https://github.com/iotaledger/identity.rs/blob/support/v0.5/bindings/wasm/examples/src/resolution.js"
-    rustGithubLink = "https://github.com/iotaledger/identity.rs/blob/support/v0.5/examples/low-level-api/resolution.rs"
-/>
+## Resolving a DID from the main network
+
+The following example demonstrates how to resolve the DID: "did:iota:H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV" from the `main` network. 
+<Tabs>
+<TabItem value="Rust" label="Rust">
+
+```rust
+use identity::iota::Resolver; 
+use identity::iota_core::IotaDID;
+use identity::iota::ResolvedIotaDocument; 
+
+let resolver: Resolver = Resolver::new().await?; 
+let did: IotaDID = IotaDID::parse("did:iota:H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV")?;
+
+let doc: ResolvedIotaDocument = resolver.resolve(&did).await?; 
+
+```
+
+</TabItem>
+<TabItem value="node" label="Node.js">
+
+```js
+const {
+    DID,
+    Resolver,
+    ResolvedDocument,
+} = require('@iota/identity-wasm/node');
+
+ const resolver = new Resolver();
+ const did = DID.parse("did:iota:H3C2AVvLMv6gmMNam3uVAjZpfkcJCwDwnZn6z3wXmqPV");
+ const doc = await resolver.resolve(did);
+
+```
+</TabItem>
+</Tabs>
+The technical details of how resolution exactly works in terms of querying the Tangle is explained in the [IOTA DID Method Specification](./../specs/did/iota_did_method_spec#read). 
```

Comment:
Link doesn't work. Can you perhaps add a high-level description along the lines of "the Tangle contains the history of the document, it is used to recreate and validate the latest state". Perhaps add any more details as you find necessary. I'd rather only people who are REALLY deep into the topic, that have to read the Method Spec.